src/main/java/redis/clients/jedis/DefaultJedisSocketFactory.java

```diff
@@ -51,6 +57,21 @@ public DefaultJedisSocketFactory(HostAndPort hostAndPort, JedisClientConfig conf
     }
   }
 
+  private void connectToFirstSuccsefulHost(Socket socket, HostAndPort hostAndPort) throws Exception {
+    List<InetAddress> hosts = Arrays.asList(InetAddress.getAllByName(hostAndPort.getHost()));
+    Collections.shuffle(hosts);
```

Comment:
Not sure whether the size of hosts is always greater than 1? If not, maybe we can do better. e.g.
```
if (hosts > 1) {
  Collections.shuffle(hosts);
}
```